llvm/test/tools/llvm-cxxfilt/quote.test

```diff
@@ -1,12 +1,21 @@
-// Show that llvm-cxxfilt --quote can emit quoted demangled symbols but not
-// without double-quoting in the case they are already quoted
+// Show that llvm-cxxfilt --quote can emit quoted demangled symbols (even at the
+// end of a line) but not without double-quoting in the case they are already
+// quoted.
+RUN: echo _Z3barv '"_Z3barv"' > %t
+RUN: echo '"_Z3barv"' _Z3barv >> %t
+// This is not mangled, thus it should not be quoted
+RUN: echo 'log()' >> %t
+// Check that an "import thunk for" prefix can be quoted along the demangled name
+RUN: echo __imp__ZSt6futureIvE >> %t
```

Comment:
Rather than use `echo`, I'd recommend the more modern `split-file` approach. In essence, you add a marker later in the file, then write the intended contents of the input file below that marker. In place of the echo lines, you simply add `RUN: split-file %s ...`. See various existing examples for correct usage.

Reply:
Interesting!